### ISSUE TITLE
What the proper length of secret keys for jwt?

### ISSUE THREAD

**Issue by TimBobkov** [mijnapp/mijnapp-frontend#40]
Hi.
I have try to startup test service for deployment. Both front and back was build successfully. But when I try to fake login I receive error:
"An unhandled exception was thrown by the application.System.ArgumentException: Specified key is not a valid size for this algorithm"
So can you provide examples of proper keys?

**Comment by versteegerik** [comment]
I think it has to be atleast 32 character, but a simple and solid key would be a guid

**Comment by TimBobkov** [comment]
I try to set all keys to 32 characters - still same error:
"2019-10-04 07:54:59.105 +00:00 [Error] Microsoft.AspNetCore.Server.IIS.Core.IISHttpServer: Connection ID "17149707382637462483", Request ID "800003d5-0000-ee00-b63f-84710c7967bb": An unhandled exception was thrown by the application.System.ArgumentException: Specified key is not a valid size for this algorithm.Parameter name: rgbKeyat Internal.Cryptography.AesImplementation.CreateTransform(Byte[] rgbKey, Byte[] rgbIV, Boolean encrypting)at Internal.Cryptography.AesImplementation.CreateEncryptor(Byte[] rgbKey, Byte[] rgbIV)
at MijnApp_Backend.Security.JwtClaimEncryptor.Encrypt(String text, String keyString) in C:\Users\azureuser\Documents\mijnapp-frontend\MijnApp-Backend\Security\JwtClaimEncryptor.cs:line 18at MijnApp_Backend.Security.JwtTokenProvider.CreateJwtSecurityToken(String bsn, SignInProvider signInProvider) in C:\Users\azureuser\Documents\mijnapp-frontend\MijnApp-Backend\Security\JwtTokenProvider.cs:line 78at MijnApp_Backend.Controllers.JwtController.SigninDigidCgiFake(IConfiguration config) in C:\Users\azureuser\Documents\mijnapp-frontend\MijnApp-Backend\Controllers\JwtController.cs:line 84at lambda_method(Closure , Object , Object[] )at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)at MijnApp_Backend.Helpers.RequestResponseLoggingMiddleware.Invoke(HttpContext context) in C:\Users\azureuser\Documents\mijnapp-frontend\MijnApp-Backend\Helpers\RequestResponseLoggingMiddleware.cs:line 23at MijnApp_Backend.Helpers.ExceptionLoggingMiddleware.Invoke(HttpContext context) in C:\Users\azureuser\Documents\mijnapp-frontend\MijnApp-Backend\Helpers\ExceptionLoggingMiddleware.cs:line 21at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.InvokeCore(HttpContext context)at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContextOfT`1.ProcessRequestAsync()"

Keys I try to use:
{
  "JwtClaimEncryption:SecretKey": "jwtclaimencryptionjwtclaimencryp",
  "Jwt:Key": "jwttestkeyjwttestkeyjwttestkeyjw",
  "DigidCgi:SharedSecret": "digidcgitestkeydigidcgitestkeydi"
}

**Comment by versteegerik** [comment]
I believe there is nothing wrong with the keys them self. I used the keys myself and it worked. 
I also used the keys in the github and they work. 

Are you using IIS to run this? If so please check https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.0&tabs=windows. We used a secrets.json, so we have locally different keys. But IIS cannot access them normally. We made a C:\UserSecrets\320e50fc-0267-417b-965b-f4c06031b254 folder, where we store this secrets.json instead of your %25USERPROFILE%25\AppData\Roaming\Microsoft\UserSecrets\320e50fc-0267-417b-965b-f4c06031b254\secrets.json folder. Maybe IIS is looking for that file and keys.

In short i believe the keys are fine. The problem is something else.
